### DISCUSSION TITLE
Sign in with credentials provider and custom login form not working

### DISCUSSION THREAD

**Discussion by mbpictures** [nextauthjs/next-auth#3534] [Help] [answered]
### Description 🐜

I've created a custom credentials provider and a custom login form, but I always get redirected to the default login form.

### Is this a bug in your own project?

Yes

### How to reproduce ☕️

Client:
```TypeScript
await signIn("credentials", {
                redirect: false,
                email: email,
                password: password
            });
```
Server:
```TypeScript
export default NextAuth({
    secret: process.env.NEXTAUTH_SECRET,
    providers: [
        CredentialsProvider({
            name: "email",
            id: "login",
            credentials: {
                email: { label: "E-Mail", type: "email", },
                password: { label: "Password", type: "password" }
            },
            async authorize(credentials, req) {
                console.log("TEST");
                const user = await prisma.adminUser.findUnique({
                    where: {
                        email: credentials.email
                    }
                });

                const checkPassword = await compare(credentials.password, user.password);
                if (!checkPassword) return null;

                return user;
            }
        }),
        CredentialsProvider({
            name: "api",
            id: "apiKey",
            credentials: {
                key: { label: "Api-Key", type: "text", },
            },
            async authorize(credentials, req) {
                const key = await hashPassword(credentials.key ?? req.headers["Api-Key"]);
                const apiKey = await prisma.adminApiKeys.findUnique({
                    where: {
                        key: key
                    },
                    include: {
                        user: true
                    }
                });

                return apiKey.user;
            }
        })
    ],
})
```

### Environment 🖥

NEXTAUTH_URL=http://localhost:3000/api/admin/auth
NEXTAUTH_SECRET=XXXXXXXXX

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment, marked as answer]
you overrode the `id`. None of the providers you registered on the server matches `credentials`

**Reply by devb** [reply]
I'm facing similar issue @balazsorban44 - Can you help me??
https://github.com/nextauthjs/next-auth/discussions/4318

**Reply by mbpictures** [reply]
@devb The ID you provide in the ```signIn``` call has to match the id of the Credentials definition in the [...nextauth] file.

**Reply by devb** [reply]
Hey @mbpictures  - this not my post.. I'm asking about this post : https://github.com/nextauthjs/next-auth/discussions/4318 << Please refer to this link and would really appreciate if you can help me on this.. I'm stucked here since last 2 days

**Comment by devb** [comment]
Can anyone help me with similar issue: https://github.com/nextauthjs/next-auth/discussions/4318  << please help me on this link
